docs/developer/index.rst

```diff
@@ -935,3 +935,187 @@ We can view the model plot object::
+  y         = Int64[3]
+  shape     = None
+  staterror = None
+  syserror  = None
+  sky       = None
+  eqpos     = None
+  coord     = logical
+
+This can then be used to evaluate a two-dimensional model,
+such as :py:class:`~sherpa.models.basic.Gauss2D`:
+
+  >>> from sherpa.models.basic import Gauss2D
+  >>> mdl = Gauss2D()
+  >>> mdl.xpos = 1500
+  >>> mdl.ypos = -100
+  >>> mdl.fwhm = 1000
+  >>> mdl.ampl = 100
+  >>> print(mdl)
+  gauss2d
+     Param        Type          Value          Min          Max      Units
+     -----        ----          -----          ---          ---      -----
+     gauss2d.fwhm thawed         1000  1.17549e-38  3.40282e+38
+     gauss2d.xpos thawed         1500 -3.40282e+38  3.40282e+38
+     gauss2d.ypos thawed         -100 -3.40282e+38  3.40282e+38
+     gauss2d.ellip frozen            0            0        0.999
+     gauss2d.theta frozen            0     -6.28319      6.28319    radians
+     gauss2d.ampl thawed          100 -3.40282e+38  3.40282e+38
+  >>> d.eval_model(mdl)
+  array([32.08564744, 28.71745887, 32.08564744])
+
+Attempting to change the coordinate system with
+:py:class:`~sherpa.astro.data.DataIMF.set_coord` will error out with a
+:py:class:`sherpa.utils.err.DataErr` instance reporting that the data
+set does not specify a shape.
+
+The shape attribute
+^^^^^^^^^^^^^^^^^^^
+
+The ``shape`` argument can be set when creating a
+:py:class:`~sherpa.astro.data.DataIMG` object to indicate that the
+data represents an "image", that is a rectangular, contiguous, set of
+pixels. It is defined as ``(nx1, nx0)``, and so matches the ndarray
+``shape`` attribute from NumPy. Operations that treat the dataset as a
+2D grid often require that the ``shape`` attribute is set.
+
+  >>> x1, x0 = np.mgridp1L4, 1:5]
+  >>> y2 = (x0 - 2.5)**2 + (x1 - 2)**2
+  >>> y = np.sqrt(y2)
+  >>> d = DataIMG('img', x0.flatten(), x1.flatten(),
+  ...             y.flatten(), shape=y.shape)
+  >>> print(d)
+  name      = img
+  x0        = Int64[12]
+  x1        = Int64[12]
+  y         = Float64[12]
+  shape     = (3, 4)
+  staterror = None
+  syserror  = None
+  sky       = None
+  eqpos     = None
+  coord     = logical
+  >>> d.get_x0()
+  array([1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4])
+  >>> d.get_x1()
+  array([1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3])
+  >>> d.get_dep()
+  array([1.80277564, 1.11803399, 1.11803399, 1.80277564, 1.5       ,
+         0.5       , 0.5       , 1.5       , 1.80277564, 1.11803399,
+         1.11803399, 1.80277564])
+  >>> d.get_axes()
+  (array([1., 2., 3., 4.]), array([1., 2., 3.]))
+  >>> d.get_dims()
+  (4, 3)
+
+Attempting to change the coordinate system with
+:py:class:`~sherpa.astro.data.DataIMF.set_coord` will error out with a
+:py:class:`sherpa.utils.err.DataErr` instance reporting that the data
+set does not contain the required coordinate system.
+
+Setting a coordinate system
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :py:class:`sherpa.astro.io.wcs.WCS` class is used to add a
```

Comment:
```suggestion
The `sherpa.astro.io.wcs.WCS` class is used to add a
```
should work, too. (Same in other places, but I'm only commenting once). One advantage is that it's a little easier to read the plaintext rst because there is less markup in it. I admit, it matter more in docstrings in the code, because those are read in plain text much more often than rst files...